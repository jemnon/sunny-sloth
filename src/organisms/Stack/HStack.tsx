import React, { FC } from 'react';
import styled from 'styled-components';
import {
  generateResponsiveProps,
  CustomInterpolation,
  SpacingProp,
  ThemeObject,
} from '../../utils/utils';
import StackProps from './stack-types';

const BaseHStack = styled.div<Pick<StackProps, 'sp'>>`
  display: flex;
  flex-direction: row;
  > * + * {
    ${({ sp = 'md1', theme: { spacing } }): CustomInterpolation =>
      generateResponsiveProps<SpacingProp>({
        cssProperty: 'margin-left',
        prop: sp,
        themeObject: spacing as ThemeObject,
      })}
  }
`;

const HStack: FC<StackProps> = ({
  as = 'div',
  dataId,
  id,
  children,
  sp = 'md1',
}) => (
  <BaseHStack as={as} data-id={dataId} id={id} sp={sp}>
    {children}
  </BaseHStack>
);

export default HStack;
