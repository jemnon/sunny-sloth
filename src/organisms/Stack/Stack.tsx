import styled from 'styled-components';
import { up } from 'styled-breakpoints';

const Stack = styled.ul`
  display: flex;
  flex-direction: column;

  margin-bottom: ${({ theme: { spacing } }): string => spacing.xlg4};
`;

export const StackItem = styled.li`
  margin-bottom: ${({ theme: { spacing } }): string => spacing.sm4};
  ${up('md')} {
    margin-bottom: ${({ theme: { spacing } }): string => spacing.md4};
  }
`;

export default Stack;
