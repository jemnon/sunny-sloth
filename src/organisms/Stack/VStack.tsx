import React, { FC } from 'react';
import styled from 'styled-components';
import {
  generateResponsiveProps,
  CustomInterpolation,
  SpacingProp,
  ThemeObject,
} from '../../utils/utils';
import StackProps from './stack-types';

const BaseVStack = styled.div<Pick<StackProps, 'sp'>>`
  display: flex;
  flex-direction: column;
  > * + * {
    ${({ sp = 'md1', theme: { spacing } }): CustomInterpolation =>
      generateResponsiveProps<SpacingProp>({
        cssProperty: 'margin-top',
        prop: sp,
        themeObject: spacing as ThemeObject,
      })}
  }
`;

const VStack: FC<StackProps> = ({
  as = 'div',
  dataId,
  id,
  children,
  sp = 'md1',
}) => (
  <BaseVStack as={as} data-id={dataId} id={id} sp={sp}>
    {children}
  </BaseVStack>
);

export default VStack;
