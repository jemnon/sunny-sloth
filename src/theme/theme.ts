const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
