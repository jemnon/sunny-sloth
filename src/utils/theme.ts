import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
