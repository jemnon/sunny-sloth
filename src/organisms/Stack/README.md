# Stack

### Description
Component description

```js
import Stack from '@fenderdigital/ui/FILL IN TYPEStack';
```

### Props
* `propName` - property name and description 
