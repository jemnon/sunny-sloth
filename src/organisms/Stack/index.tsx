import Stack, { StackItem } from './Stack';

export { StackItem };
export default Stack;
