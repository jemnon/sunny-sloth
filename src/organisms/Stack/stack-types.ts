import { ElementType, ReactNode } from 'react';
import { SpacingProp } from '../../utils/utils';
import { BaseProps } from '../../types/base-props';

type BasePropsType = Pick<BaseProps, 'dataId' | 'id'>;

interface StackProps extends BasePropsType {
  as?: ElementType;
  children: ReactNode;
  sp?: SpacingProp;
}

export default StackProps;
