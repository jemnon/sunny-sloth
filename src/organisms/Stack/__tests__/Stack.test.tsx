import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Stack from '../Stack';
import theme from '../../../theme';

describe('<Stack />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Stack />
    </ThemeProvider>
  );
  it('exists', (): void => {
    expect(Stack).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});