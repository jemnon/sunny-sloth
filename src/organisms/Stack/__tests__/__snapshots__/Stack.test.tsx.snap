// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stack /> renders defaults correctly 1`] = `
<ul
  className="sc-bdfBwQ dcTsLa"
/>
`;
